srcpkgs/docker/template

```diff
@@ -1,138 +1,10 @@
-		SET(CMAKE_CROSSCOMPILING TRUE)
-
-		SET(CMAKE_SYSTEM_PROCESSOR ${_CMAKE_SYSTEM_PROCESSOR})
-
-		SET(CMAKE_FIND_ROOT_PATH  ${XBPS_CROSS_BASE})
-
-		SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-		SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
-		SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
-		_EOF
-		cmake_args+=" -DCMAKE_TOOLCHAIN_FILE=cross_${XBPS_CROSS_TRIPLET}.cmake"
-	fi
-	CFLAGS="${CFLAGS/ -pipe / }" CXXFLAGS="${CXXFLAGS/ -pipe / }" \
-	cmake ${cmake_args} ..
-
-	mkdir -p $wrksrc/gopath/src/github.com/docker
-
-	msg_normal "Preparing docker-proxy import path\n"
-	cd $wrksrc/libnetwork-$_libnetwork
-	rm -f $wrksrc/gopath/src/github.com/docker/libnetwork
-	ln -s "$PWD" $wrksrc/gopath/src/github.com/docker/libnetwork
-
-	msg_normal "Preparing docker-cli import path\n"
-	cd $wrksrc/docker-ce-$version
-	rm -f $wrksrc/gopath/src/github.com/docker/cli
-	ln -s "$PWD/components/cli" $wrksrc/gopath/src/github.com/docker/cli
-	rm -f $wrksrc/gopath/src/github.com/docker/docker
-	ln -s "$PWD/components/engine" $wrksrc/gopath/src/github.com/docker/docker
-
-	BUILDTIME=$(date --utc --rfc-3339 ns)
-	BUILDTIME=${BUILDTIME/ /T}
-	vsed -i -e "
-		/GitCommit/s/\".*\"/\"v$version\"/;
-		/Version/s/\".*\"/\"$version\"/;
-		/BuildTime/s/\".*\"/\"$BUILDTIME\"/;
-	" \
-		components/engine/dockerversion/version_lib.go \
-		components/cli/cli/version/version.go
-}
-
-do_build() {
-	export GOPATH="$wrksrc/gopath"
-	mkdir -p $wrksrc/gopath/bin
-
-	msg_normal "Building docker-init\n"
-	cd $wrksrc/tini-$_tini_version/build
-	make tini-static
-
-	msg_normal "Building docker-proxy\n"
-	go build -buildmode=pie \
-		-ldflags="$go_ldflags -linkmode=external" \
-		-o $wrksrc/gopath/bin/docker-proxy \
-		github.com/docker/libnetwork/cmd/proxy
-
-	msg_normal "Building docker cli\n"
-	go build -buildmode=pie \
-		-ldflags="$go_ldflags" -o $wrksrc/gopath/bin/docker \
-		github.com/docker/cli/cmd/docker
-
-	msg_normal "Building docker daemon\n"
-	go build -buildmode=pie \
-		-tags 'libdm_dlsym_deferred_remove seccomp apparmor' \
-		-ldflags="$go_ldflags" \
-		-o $wrksrc/gopath/bin/dockerd \
-		github.com/docker/docker/cmd/dockerd
-}
-
-do_install() {
-	vbin tini-$_tini_version/build/tini-static docker-init
-	vlicense tini-$_tini_version/LICENSE docker-init.MIT
-	vbin gopath/bin/docker-proxy
-	vbin gopath/bin/docker
-	vbin gopath/bin/dockerd
-	cd docker-ce-$version
-	vcompletion components/cli/contrib/completion/bash/docker bash
-	vcompletion components/cli/contrib/completion/zsh/_docker zsh
-	vcompletion components/cli/contrib/completion/fish/docker.fish fish
-	vinstall components/engine/contrib/syntax/vim/syntax/dockerfile.vim 644 \
-		usr/share/vim/vimfiles/syntax/dockerfile.vim
-	vinstall components/engine/contrib/syntax/vim/ftdetect/dockerfile.vim 644 \
-		usr/share/vim/vimfiles/ftdetect/dockerfile.vim
-	vlicense components/engine/contrib/syntax/vim/LICENSE dockerfile.vim.BSD-2-Clause
-	vsv docker
-}
+build_style=meta
+depends="docker-engine docker-cli"
```

Comment:
If `docker-cli` should depend on `moby` (see above), then `depends="docker-cli"` would be appropriate here.

Reply:
See my larger response.